libr/bin/p/bin_mach064.c

```diff
@@ -5,11 +5,61 @@
 
 #include "objc/mach064_classes.h"
 
+static bool is_kernelcache(struct MACH0_(obj_t) *mach0) {
+	struct section_t *sections = NULL;
+	if (!(sections = MACH0_(get_sections) (mach0))) {
+		return false;
+	}
+
+	int incomplete = 3;
+	int i = 0;
+	for (; !sections[i].last; i++) {
+		if (strstr (sections[i].name, "__PRELINK_INFO.__info")) {
+			if (!--incomplete) {
+				break;
+			}
+		}
+
+		if (strstr (sections[i].name, "__PRELINK_TEXT.__text")) {
+			if (!--incomplete) {
+				break;
+			}
+		}
+
+		if (strstr (sections[i].name, "__PRELINK_DATA.__data")) {
+			if (!--incomplete) {
+				break;
+			}
+		}
+	}
+
+	R_FREE (sections);
+
+	return incomplete == 0;
+}
+
 static bool check_bytes(const ut8 *buf, ut64 length) {
 	if (buf && length > 4) {
 		if (!memcmp (buf, "\xfe\xed\xfa\xcf", 4) ||
 			!memcmp (buf, "\xcf\xfa\xed\xfe", 4)) {
-			return true;
+			RBuffer *tbuf = r_buf_new ();
+			if (!tbuf) {
+				return false;
+			}
+			struct MACH0_(obj_t) *test = NULL;
+			bool regular_mach0 = true;
+			r_buf_set_bytes (tbuf, buf, length);
```

Comment:
ouch, thats a bad idea, the check byets should just check, not fill the whole buffer, also assuming the buf+len will cover the whole file is not correct. we shold only use RBuffer in rbin plugins

Reply:
ok reverting for now - then we'll have to find a way to fail here if it's a kernelcache, so we can avoid to use `-F kernelcache` all the time.

Reply:
is there an amount of bytes it's guaranteed to have here?

Reply:
nope, you must use the r_buf_read_at to read the data you want , there's an implicit memory copy but it's the only way to be safe in case the backend doesnt provides a linear buffer